### ISSUE TITLE
Add the ability to pass a reviver function to Http Response's Json method

### ISSUE THREAD

**Issue by TFleury** [angular/http#84]
`Json.Parse`  method supports passing a reviver function. It's especially useful to correctly handle date parsing from JSON.

**Comment by robwormald** [comment]
Interesting, I'd entirely forgotten this was a thing. In theory, its easily doable, but one thing I'd like to be able to do with http eventually is to leverage the async JSON parser from the fetch API, which doesn't support this functionality (because the parsing happens in C++ land) - see https://github.com/whatwg/fetch/issues/104 for a bit more detail on that.

**Comment by templth** [comment]
Good idea! I created a PR for this issue: https://github.com/angular/angular/pull/7369. Feel free to make me your comment...

**Comment by mmc41** [comment]
@templth Has this feature gone/moved or is it missing a decl. In angular 2 final, the json() method is on the Body class and here there is no way to set a reviver ?
